llama-index-core/llama_index/core/workflow/workflow.py

```diff
@@ -113,20 +122,23 @@ async def _task(
                         print(f"Running step {name}")
 
                     # run step
-                    args = []
-                    if config.pass_context:
-                        args.append(self.get_context(name))
-                    args.append(ev)
+                    kwargs = {}
```

Comment:
Moving to passing data to steps as keyword arguments, this will avoid problems like "in which position is the context supposed to be". Added in this PR to remove the need of passing a list of service names to `step()`, now the parameter name and type will determine if a step requires a service.